Userland/Libraries/LibGUI/FileIconProvider.h

```diff
@@ -1,29 +1,57 @@
 /*
  * Copyright (c) 2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Mustafa Quraish <mustafa@serenityos.org>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #pragma once
 
 #include <AK/Forward.h>
+#include <LibCore/Object.h>
 #include <LibGUI/Forward.h>
+#include <LibGUI/Icon.h>
 #include <sys/types.h>
 
 namespace GUI {
 
-class FileIconProvider {
+class FileIconProvider final : public Core::Object {
+    C_OBJECT(FileIconProvider);
```

Comment:
This doesn't need to be a Core::Object (for that matter, a singleton refcounted object is weird anyway...) at all.

Reply:
Fair enough, I'll make the change. I had no idea how singletons work in the codebase so I just ended up [copying it from FileSystemAccessClient](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibFileSystemAccessClient/Client.cpp#L18-L25) 😅

I'm assuming that's different because it's inheriting from the IPC stuff and what not. I'll just make it a simple static variable in that case.